### PR TITLE
Chapter1/homework/4

### PR DESCRIPTION
**Pull request by pinology**

### CODE REVIEW

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            {
+                isWeightValid = false;
+            }
+
+            if (height == 0 || height == -1)
+            {
+                isHeightValid = false;
+            }
+
+            if (!isWeightValid || !isHeightValid)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199&bid=16273047)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        {
+            bool isWeightValid = true;
+            bool isHeightValid = true;
+
+            if (weight == 0 || weight == -1)
+            {
+                isWeightValid = false;
+            }
+
+            if (height == 0 || height == -1)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                return InvalidBMI(isWeightValid, isHeightValid, weight, height);
+            }
+
+            else
+            {
+                return weight / (height * height);
+            }
+        }
+
+        public static float InvalidBMI(bool isWeightValid, bool isHeightValid, float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Rename method 'InvalidBMI' to match pascal case naming rules, consider using 'InvalidBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                }
+            }
+        }
+
+        public static float CalculateBodyMassIndex(float weight, float height)
+        {
+            bool isWeightValid = true;
+            bool isHeightValid = true;
+
+            if (weight == 0 || weight == -1)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

File: Src/BootCamp.Chapter/Checks.cs [hunk 5]
```diff
@@ -16,26 +16,24 @@ public static class Checks
...
         public static int PromptInt(string message)
         {
-            // To do: call your implementation. 
-            return 0;
+            return Lesson4.PromptInt(message);
         }
 
         public static string PromptString(string message)
         {
             // To do: call your implementation.
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove the comments, which are now redundant.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 6]
```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Globalization;
+
+namespace BootCamp.Chapter
+{
+    public class Lesson4
+    {
+        public static void Demo()
+        {
+            //run twice
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Redundant comment- not needed.

Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: Write code that speaks for itself, do not use comments for explaining blocks of code.
Use step functions to explain blocks of code.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            for (int i = 0; i < 2; i++)
+            {
+                string firstName = PromptString("Input First Name: ");
+                string lastName = PromptString("Input Last Name: ");
+                int age = PromptInt("Input age: ");
+                float weight = PromptFloat("Input weight (in kg): ");
+                float height = PromptFloat("Input height (in m): ");
+
+                //print info
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Please remove redundant comment.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+                Console.WriteLine(firstName + " " + lastName + " is " + age + " years old, his weight is " + weight + " kg and his height is " + height + " m.");
+
+                //calculate and print bmi
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Please remove redundant comment.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                float bmi = CalculateBodyMassIndex(weight, height);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: CalculateBmi

File: Src/BootCamp.Chapter/Lesson4.cs (hunk 4 above)
Comment (Almantask) [reply Src/BootCamp.Chapter/Lesson4.cs]: For floating point numbers use a range check, otherwise it might not work.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199&bid=16273047)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+            {
+                isWeightValid = false;
+            }
+
+            if (height <= 0)
+            {
+                isHeightValid = false;
+            }
+
+            if (!isWeightValid || !isHeightValid)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson4.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199&bid=16273047)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+                    return -1;
+                }
+                else
+                {
+                    return returnFloat;
+                }
+            }
+        }
+
+        public static float CalculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+        {
+            if (weight <= 0 || height <= 0)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+                return InvalidBMI(weight, height);
+            }
+
+            else
+            {
+                return weight / (height * height);
+            }
+        }
+
+        public static float InvalidBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Codacy Issue found: [Rename method 'InvalidBMI' to match pascal case naming rules, consider using 'InvalidBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4917199)

**Comment by pinology** [comment]
hello... could any mentor help me out here? is there anything i should change?

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+namespace BootCamp.Chapter
+{
+    public class Lesson4
+    {
+        public static void Demo()
+        {
+            for (int i = 0; i < 2; i++)
+            {
+                string firstName = PromptString("Input First Name: ");
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: Contents of for block could be moved out to a function.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+        public static int PromptInt(string checkString)
+        {
+            Console.Write(checkString);
+            string returnString = Console.ReadLine();
+            if (returnString == "")
+            {
+                return 0;
+            }
+            else
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: No need else statement.

File: Src/BootCamp.Chapter/Lesson4.cs [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+        }
+        public static float PromptFloat(string checkString)
+        {
+            Console.Write(checkString);
+            string returnString = Console.ReadLine();
+            if (returnString == "")
+            {
+                return 0;
+            }
+            else
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson4.cs]: No need else

**Comment by pinology** [comment]
I've read the comments and appreciate them! I will keep them in mind, thank you :)

**Pull request closed without merging**